drivers/clock_control/clock_control_nrf2_common.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_CLOCK_CONTROL_NRF2_COMMON_H_
+#define ZEPHYR_DRIVERS_CLOCK_CONTROL_NRF2_COMMON_H_
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/onoff.h>
+
+#define FLAGS_COMMON_BITS 10
+
+struct clock_onoff {
+	struct onoff_manager mgr;
+	onoff_notify_fn notify;
+	uint8_t idx;
+};
+
+/**
+ * @brief Defines a type for specific clock configuration structure.
+ *
+ * @param type suffix added clock_config_ to form the type name.
+ * @param _onoff_cnt number of clock configuration options to be handled;
+ *                   for each one a separate onoff manager instance is used.
+ */
+#define NRF2_STRUCT_CLOCK_CONFIG(type, _onoff_cnt) \
```

Comment:
This is not a public API, so I don't think we need to use these `nrf2_` prefixes here (which IMO look a bit weird without `clock_control_` before them).

Reply:
done